File: python/models/Day.py
from . import Bitcoin

class Day:
    def __init__(self, bitcoin):
        self.day = bitcoin.timestamp
        self.bitcoins = []
        self.average_price = float()
        self.number_of_bitcoin = 0
        self.label = None

    def add_bitcoin(self, bitcoin):
        self.bitcoins.append(bitcoin)
        self.number_of_bitcoin = len(self.bitcoins)

    def calculate_average_of_bitcoin(self):
        self.average_price = sum(bitcoin.weighted_price for bitcoin in self.bitcoins)/len(self.bitcoins)

    def __str__(self):
        return_string =  str(self.day) + "," + str('%.15f' % self.average_price).strip("0") + "," 
        if self.label is not None:
            return_string += str(self.label)
        return return_string

File: java/model/models/Day.java
package models;

import java.util.ArrayList;
import java.util.List;

public class Day {

  String timestamp;
  double averagePrice;
  List<Bitcoin> bitcoins;
  float label;
  int numberOfBitcoin;


  public Day(Bitcoin bitcoin) {
    this.timestamp = bitcoin.getTimestamp();
    bitcoins = new ArrayList<>();
    this.label = -1;
    this.numberOfBitcoin = 0;
  }

  public void addBitcoin(Bitcoin bitcoin) {
    this.bitcoins.add(bitcoin);
    this.numberOfBitcoin = this.bitcoins.size();
  }

  public void calculateAveragePrice() {
    double sum_of_prices = 0;
    for (Bitcoin bitcoin : this.bitcoins) {
      sum_of_prices += bitcoin.getWeightPrice();
    }
    averagePrice = sum_of_prices / bitcoins.size();
  }

  public String getTimestamp() {
    return timestamp;
  }

  public void setLabel(float label) {
    this.label = label;
  }

  public double getAveragePrice() {
    return averagePrice;
  }

  @Override
  public String toString() {
    if (this.label != -1)
      return timestamp + "," + averagePrice + "," + label + '\n';
    else
      return timestamp + "," + averagePrice + "," + '\n';
  }
}
